app_dart/lib/src/request_handlers/authorize_agent.dart

```diff
@@ -0,0 +1,69 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:cocoon_service/src/model/appengine/agent.dart';
+import 'package:cocoon_service/src/request_handling/exceptions.dart';
+import 'package:cocoon_service/src/service/agent_service.dart';
+import 'package:gcloud/db.dart';
+import 'package:meta/meta.dart';
+
+import '../datastore/cocoon_config.dart';
+import '../request_handling/api_request_handler.dart';
+import '../request_handling/authentication.dart';
+import '../request_handling/body.dart';
+import '../service/agent_service.dart';
+import '../service/datastore.dart';
+
+@immutable
+class AuthorizeAgent extends ApiRequestHandler<AuthorizeAgentResponse> {
+  const AuthorizeAgent(
+    Config config,
+    AuthenticationProvider authenticationProvider, {
+    @visibleForTesting
+        this.datastoreProvider = DatastoreService.defaultProvider,
+  }) : super(config: config, authenticationProvider: authenticationProvider);
+
+  final DatastoreServiceProvider datastoreProvider;
+
+  static const String agentIdParam = 'AgentID';
+
+  @override
+  Future<AuthorizeAgentResponse> post() async {
+    checkRequiredParameters(<String>[agentIdParam]);
+
+    final String agentId = requestData[agentIdParam];
+    final DatastoreService datastore = datastoreProvider();
+    final AgentService agentService = AgentService();
```

Comment:
This should be injected so that it can be mocked out in tests.

Reply:
Revised following style of datastore.